src/transformStateWithClones.js

```diff
@@ -7,7 +7,39 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const copy = [];
+
+  actions.forEach(action => {
+    if (action.type === 'addProperties') {
+      const add = {
+        ...(copy.length
+          ? copy[copy.length - 1]
+          : state),
+        ...action.extraData,
+      };
```

Comment:
it can be simplified 
```suggestion
      Object.assign(stateCopy, action.extraData);
```